darts/timeseries.py

```diff
@@ -1606,36 +1650,62 @@ def pd_dataframe(self, copy=True, suppress_warnings=False) -> pd.DataFrame:
 
             comp_name = list(self.components)
             samples = range(self.n_samples)
-            df_col_names = [
+            columns = [
                 "_s".join((comp_name, str(sample_id)))
                 for comp_name, sample_id in itertools.product(comp_name, samples)
             ]
-
-            if copy:
-                return pd.DataFrame(
-                    self._xa.stack(data=(DIMS[1], DIMS[2])).values.copy(),
-                    index=self._time_index.copy(),
-                    columns=df_col_names.copy(),
-                )
-            else:
-                return pd.DataFrame(
-                    self._xa.stack(data=(DIMS[1], DIMS[2])).values,
-                    index=self._time_index,
-                    columns=df_col_names,
-                )
+            data = self._xa.stack(data=(DIMS[1], DIMS[2]))
         else:
-            if copy:
-                return pd.DataFrame(
-                    self._xa[:, :, 0].values.copy(),
-                    index=self._time_index.copy(),
-                    columns=self._xa.get_index(DIMS[1]).copy(),
-                )
-            else:
-                return pd.DataFrame(
-                    self._xa[:, :, 0].values,
-                    index=self._time_index,
-                    columns=self._xa.get_index(DIMS[1]),
-                )
+            data = self._xa[:, :, 0].values
+            columns = self._xa.get_index(DIMS[1])
+
+        time_index = self._time_index
+
+        if time_as_index:
+            # special path for pandas with index
+            import pandas as pd
+
+            return pd.DataFrame(data=data, index=time_index, columns=columns)
+
+        data_ = {
+            time_index.name: time_index,  # set time_index as left-most column
+            **{col: data[:, idx] for idx, col in enumerate(columns)},
+        }
```

Comment:
you can overwrite `data`

```suggestion
        data = {
            time_index.name: time_index,  # set time_index as left-most column
            **{col: data[:, idx] for idx, col in enumerate(columns)},
        }
```